server/src/Domain/Entities/Lecture.cs

```diff
@@ -1,9 +1,19 @@
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using StudyZen.Domain.Constraints;
+
 namespace StudyZen.Domain.Entities;
 
 public sealed class Lecture : BaseEntity
 {
-    public int CourseId { get; }
+    public int CourseId { get; set; }
```

Comment:
https://www.entityframeworktutorial.net/code-first/foreignkey-dataannotations-attribute-in-code-first.aspx